cli/test/test_cmdskip.py

```diff
@@ -0,0 +1,103 @@
+from click.testing import CliRunner
+import unittest
+from popper_test import PopperTest
+import popper.commands.cmd_run as run
+import unittest
+import os
+import subprocess
+import sys
+
+FIXDIR = f'{os.path.dirname(os.path.realpath(__file__))}/fixtures'
+
+
+def _wfile(name, format):
+    return f'{FIXDIR}/{name}.{format}'
+
+
+class TestSkip(unittest.TestCase, PopperTest):
+
+    def test_skip(self):
+
+        workflow_file_loc = _wfile("skip", "workflow")
+
+        runner = CliRunner()
+        result = runner.invoke(
+            run.cli, ['a1', '--wfile', workflow_file_loc,
+                      '--skip', 'a2'])
+        assert result.exit_code == 1
+
+        result = runner.invoke(
+            run.cli, ['--dry-run', '--wfile', workflow_file_loc,
+                      'a1'])
+        assert result.exit_code == 0
+
+        result = runner.invoke(
+            run.cli, ['--dry-run', '--wfile', workflow_file_loc,
+                      'a1'])
+        assert result.exit_code == 0
+
+        result = runner.invoke(
+            run.cli, ['--dry-run', '--wfile', workflow_file_loc,
+                      'a2'])
+        assert result.exit_code == 0
+
+        result = runner.invoke(
+            run.cli, ['--dry-run', '--wfile', workflow_file_loc,
+                      'b'])
+        assert result.exit_code == 0
+
+        result = runner.invoke(
+            run.cli, ['--dry-run', '--wfile', workflow_file_loc,
+                      'c'])
+        assert result.exit_code == 0
+
+        result = runner.invoke(
+            run.cli, ['--dry-run', '--wfile', workflow_file_loc,
+                      'd'])
+        assert result.exit_code == 0
+
+        result = runner.invoke(
+            run.cli, ['--dry-run', '--wfile', workflow_file_loc,
+                      '--skip', 'e1'])
+        assert result.exit_code == 1
+
+        result = runner.invoke(
+            run.cli, ['--wfile', workflow_file_loc, '--dry-run', '--skip',
+                      'a1', '--skip', 'a2'])
+        assert result.exit_code == 1
+
+        result = self.check_output(['popper', 'run', '--dry-run', '--wfile',
+                                    workflow_file_loc, '--skip', 'a1'],
+                                   ["a2", "b", "c", "d"])
+        assert result == 0
+
+        result = self.check_output(
+            ['popper', 'run', '--wfile', workflow_file_loc, '--dry-run',
+             '--skip', 'a2'], ['a1', 'b', 'c', 'd'])
+        assert result == 0
+
+        result = runner.invoke(
+            run.cli, ['--wfile', workflow_file_loc, '--dry-run',
+                      '--skip', 'b', '--skip', 'c'])
+        assert result.exit_code == 1
```

Comment:
we might want to also check that the error being reported is the one we expect.

Reply:
Yes, It would be nice to have it since we can debug the errors in a better way, Thanks for pointing it out.

Reply:
Since the output is blank so should I match using subprocess only?

Reply:
we can check the contents of the loggers, so no need to use subprocess